From 035ad708efb4f17860b76f771c35e49fc7f797f1 Mon Sep 17 00:00:00 2001
From: Dirk Groeneveld <dirkg@allenai.org>
Date: Tue, 30 Nov 2021 10:44:26 -0800
Subject: [PATCH 4/4] Makes TensorCache work with Python 3.9

---
 allennlp/common/file_utils.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/allennlp/common/file_utils.py b/allennlp/common/file_utils.py
index 9660eeb9..d0efd17c 100644
--- a/allennlp/common/file_utils.py
+++ b/allennlp/common/file_utils.py
@@ -596,9 +596,7 @@ class TensorCache(MutableMapping[str, Tensor], ABC):
         except FileNotFoundError:
             result = None
         if result is None:
-            result = super(TensorCache, cls).__new__(
-                cls, filename, read_only=read_only, **kwargs
-            )  # type: ignore
+            result = super(TensorCache, cls).__new__(cls)
         return result
 
     def __init__(
-- 
2.32.0.windows.2

